src-java/stats-topology/stats-storm-topology/src/test/java/org/openkilda/wfm/topology/stats/StatsTopologyTest.java

```diff
@@ -426,6 +494,519 @@ public void flowStatsTest() {
+        validateHaFlowMeterStats(forward, haFlow.getForwardPath().getCookie(), SWITCH_ID_1,
+                METER_ID_HA_FLOW_PATH_FORWARD.getValue(), SUB_FLOW_ID_SHARED, false);
+
+        //reverse
+        MeterStatsEntry reverse1 = new MeterStatsEntry(
+                METER_ID_HA_FLOW_PATH_REVERSE.getValue(), 400L, 500L);
+        sendStatsMessage(new MeterStatsData(SWITCH_ID_2, Collections.singletonList(reverse1)));
+        validateHaFlowMeterStats(reverse1, haFlow.getReversePath().getCookie(),
+                SWITCH_ID_2, METER_ID_HA_FLOW_PATH_REVERSE.getValue(), null, true);
+    }
+
+    /**
+     * 2 switch ha-flow flow-stats test.
+     * 1----2   where 1 is an y-point switch and 2 is an egress for forward direction
+     */
+    @Test
+    public void haFlowStatsTest5() {
+        HaFlow haFlow = createHaFlow5();
+        sendUpdateHaFlowPathInfo(haFlow);
+        //forward
+        FlowStatsEntry forwardYPoint = new FlowStatsEntry(1, haFlow.getForwardPath().getCookie().getValue(),
+                150L, 300L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_1, Collections.singletonList(forwardYPoint)));
+        validateHaFlowStats(forwardYPoint, haFlow.getForwardPath().getCookie(), SWITCH_ID_1, true,
+                true, true, SUB_FLOW_ID_SHARED);
+
+        FlowStatsEntry forwardEgress = new FlowStatsEntry(1, haFlow.getForwardPath().getCookie().getValue(),
+                150L, 300L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_2, Collections.singletonList(forwardEgress)));
+        validateHaFlowStats(forwardEgress, haFlow.getForwardPath().getCookie().toBuilder()
+                        .subType(FlowSegmentCookie.FlowSubType.HA_SUB_FLOW_1).build(), SWITCH_ID_2, false,
+                true, true, SUB_FLOW_ID_SHARED);
+
+
+        //reverse:
+        FlowSegmentCookie cookieReverse = haFlow.getReversePath().getCookie();
+
+        FlowStatsEntry reverseYPoint1 = new FlowStatsEntry(1,
+                cookieReverse.toBuilder().subType(FlowSegmentCookie.FlowSubType.HA_SUB_FLOW_1).build().getValue(),
+                150L, 300L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_2, Collections.singletonList(reverseYPoint1)));
+        validateHaFlowStats(reverseYPoint1, cookieReverse.toBuilder()
+                        .subType(FlowSegmentCookie.FlowSubType.HA_SUB_FLOW_1).build(), SWITCH_ID_2, true,
+                false, true, SUB_FLOW_ID_1);
+
+        FlowStatsEntry reverseYPoint2 = new FlowStatsEntry(1,
+                cookieReverse.toBuilder().subType(FlowSegmentCookie.FlowSubType.HA_SUB_FLOW_2).build().getValue(),
+                150L, 300L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_2, Collections.singletonList(reverseYPoint2)));
+        validateHaFlowStats(reverseYPoint2, cookieReverse.toBuilder()
+                        .subType(FlowSegmentCookie.FlowSubType.HA_SUB_FLOW_2).build(), SWITCH_ID_2, true,
+                false, true, SUB_FLOW_ID_2);
```

Comment:
same

Reply:
is it still relevant for the current test's topology?